File: 392.is-subsequence.cpp
/*
 * @lc app=leetcode id=392 lang=cpp
 *
 * [392] Is Subsequence
 */

#include <bits/stdc++.h>
using namespace std;

// @lc code=start
class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        if (s.size() == 0)
            return true;

        int j = 0;

        for (int i = 0; i < t.size(); ++i)
        {
            if (s[j] == t[i])
            {
                j++;
            }

            if (j == s.size())
            {
                return true;
            }
        }

        return false;
    }
};
// @lc code=end

File: 392.is-subsequence.py
#
# @lc app=leetcode id=392 lang=python3
#
# [392] Is Subsequence
#

# @lc code=start
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = 0
        if len(s) == 0:
            return True
        if len(t) < len(s):
            return False
        
        for j in range(len(t)):
            if t[j] == s[i]:
                i += 1
            if i == len(s):
                return True
        return False
        
        
# @lc code=end

